### PR TITLE
chore(docs): update readme and migration guide

### PR DESCRIPTION
**Pull request by marcolink**
# Description
Include all-new features to readme and advanced docs. Additionally define breaking changes and migration plan. Introduces a new readme file dedicated to typescript. 

**Changes to the header for each `md` file:**
- re-design the look
- introduce navigation to switch between files. 

### Rendered docs: 
- [README.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/README.md)
- [ADVANCED.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/ADVANCED.md)
- [MIGRATION.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/MIGRATION.md)
- [TYPESCRIPT.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/TYPESCRIPT.md)

# Questions
- ~~Is [SETUP.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/SETUP.md) still relevant?~~ decided to remove them
- ~~Is [SECURITY.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/SECURITY.md) still relevant?~~ decided to remove them

### CODE REVIEW

File: TYPESCRIPT.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    'fields.price[in]': [10, 20]
+})
+```
+
+#### Limitation
+Due to the limitation with recursive types, we can only calculate keys on the root level of your content type.
+There is currently no way to calculate keys for nested (recursive) content types.
+
+## Response types
+With version `10.0.0` we introduce chained clients to make better assumptions on response types.
```
Comment (ruderngespra) [inline TYPESCRIPT.md:111]: Maybe add a link to the explanation in the default Readme (https://github.com/contentful/contentful.js/blob/typescript/update-documentation/README.md#chained-clients)?

File: TYPESCRIPT.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+{
+  "fields": {
+    "productName": {
+      "de-DE": "<field-value>",
+      "en-US": "<field-value>"
+    }
+  }
+}
+```
+
```
Comment (ruderngespra) [inline TYPESCRIPT.md:141]: Should we also have sections for `withoutLinkResolution` and `withoutUnresolvableLinks`?

File: MIGRATION.md [hunk 3]
```diff
@@ -19,6 +50,37 @@ From version 3.0.0 onwards, you can access documentation for a specific version
...
+#### Example
+```js
+const entriesWithoutResolvedLinks = await client.withoutLinkResolution.getEntries()
+```
+```js
+const entriesWithoutUnresolvableLinks = await client.withoutUnresolvableLinks.getEntries()
+```
+
+### Query param `locale='*'`
+The query param `locale='*'` has been deprecated in favor of the client chain `withALlLocales`. This is due to its special response format.
```
Comment (ruderngespra) [inline MIGRATION.md:79]:
```suggestion
The query param `locale='*'` has been deprecated in favor of the client chain `withAllLocales`. This is due to its special response format.
```

**Review by ruderngespra — CHANGES_REQUESTED** [review]
Really nice :) Some suggestions / questions / typo corrections, but all in all lgtm. 🎉 

> Is [SETUP.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/SETUP.md) still relevant?
> Is [SECURITY.md](https://github.com/contentful/contentful.js/blob/typescript/update-documentation/SECURITY.md) still relevant?

No to both, if we see confusions related to the information in those files, let's reintroduce them, but for now I would remove them.

File: TYPESCRIPT.md (hunk 2 above)
Comment (marcolink) [reply TYPESCRIPT.md:142]: @ruderngespra yes, I added both

File: MIGRATION.md [hunk 4]
```diff
@@ -19,6 +50,37 @@ From version 3.0.0 onwards, you can access documentation for a specific version
...
+
+#### Example
+```js
+const entriesWithoutResolvedLinks = await client.withoutLinkResolution.getEntries()
+```
+```js
+const entriesWithoutUnresolvableLinks = await client.withoutUnresolvableLinks.getEntries()
+```
+
+### Query param `locale='*'`
```
[the remark this answers is not in the document]
Comment (marcolink) [reply MIGRATION.md:77]: @ruderngespra Good catch! I added a note on the `resolveLinks` query params 👍

**Review by ruderngespra — APPROVED** [review]

File: TYPESCRIPT.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+### `withoutLinkResolution`
+If the current chain includes `withoutLinkResolution`, the returned type doesn't define linked entities as Link objects.
```
Comment (damienxy) [inline TYPESCRIPT.md:144]: Do you mean the opposite? With `withoutLinkResolution` linked entities stay Link objects (as opposed to being resolved)

File: TYPESCRIPT.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    "productName": {
+      "type": "Link",
+      "linkType": "Entry",
+      "id": "linkedProductId" 
+    }
+  }
+}
+```
+### `withoutUnresolvableLinks`
+If the current chain includes `withoutUnresolvableLinks`, the returned type doesn't include linked fields that are not resolvable.
```
Comment (damienxy) [inline TYPESCRIPT.md:175]:
```suggestion
If the current chain includes `withoutUnresolvableLinks`, the returned type doesn't include linked fields that are not resolvable, for example if the linked entity does not exist anymore or is not yet published.
```

File: TYPESCRIPT.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+```json
+{
+  "fields": {}
+}
+```
+
+
+## Generating type definitions for content types
+It is recommended to define field types for all your content types. This helps the type system to infer all possible query keys/value types for you.
+There are several OSS projects out there to generate type definitions for content types:
```
Comment (damienxy) [inline TYPESCRIPT.md:203]:
```suggestion
There are several OSS projects out there to generate type definitions for Contentful content types:
```

File: MIGRATION.md [hunk 8]
```diff
@@ -19,6 +48,55 @@ From version 3.0.0 onwards, you can access documentation for a specific version
 
 You can upgrade to a major version using `npm update contentful`
 
+## Migration to contentful.js 10.x
+Version `10.0.0` is a complete rewrite in Typescript. This version introduces a new concept of [chained clients](#chained-clients).
```
Comment (damienxy) [inline MIGRATION.md:52]: Wrong link to chained clients

File: MIGRATION.md [hunk 9]
```diff
@@ -19,6 +48,55 @@ From version 3.0.0 onwards, you can access documentation for a specific version
...
+- Edge >= 80,
+- Firefox >= 75,
+- Safari >= 13,
+
+We completely dropped support for old IE browsers (no `legacy` bundle)
+> You can always find the supported browsers in our shared [browserlist-config](https://github.com/contentful/browserslist-config/blob/master/index.js)
+
+### Client config
+The fields `resolveLinks` and `removeUnresolved` have been removed from the TS config interface for `createClient`. 
+We still respect them for now, but you will see a warning in the console when using them. Instead, you should use one of the [chained clients](./README.md#chained-clients) to achieve the same result.
```
Comment (damienxy) [inline MIGRATION.md:66]: This text should be updated (we no longer support them), and I would also mention that it's a breaking change (maybe a "Breaking changes" section?)

File: MIGRATION.md [hunk 10]
```diff
@@ -19,6 +48,55 @@ From version 3.0.0 onwards, you can access documentation for a specific version
...
+We kept the defaults. If you want to get a response without resolved links, please use the client chain `withoutLinkResolution`.
```
Comment (damienxy) [inline MIGRATION.md:67]:
```suggestion
We kept the defaults (linked entities are by default resolved, and, if unresolvable, represented as a Link object). If you want to get a response without resolved links, please use the client chain `withoutLinkResolution`.
```

File: README.md [hunk 11]
```diff
@@ -211,174 +221,70 @@ contentful.createClient({
...
+
+This way, we make developing with `contentful.js` much more predictable and safer.
+
+When initialising a client, you will receive an instance of the [`DefaultClient`](lib/create-contentful-api.ts#L91) shape.
+
+| Chain                      | Modifier                                                                        |
+|----------------------------|---------------------------------------------------------------------------------|
+| default                    | with link resolution, including unresolvable links, single locale               |
+| `withAllLocales`           | entries for all locales                                                         |
+| `withoutLinkResolution`    | inline linked entries                                                           |
```
[the remark this answers is not in the document]
Comment (marcolink) [reply README.md]: @ruderngespra @damienxy added a new section "Link Resolution"

File: README.md [hunk 12]
```diff
@@ -211,174 +219,70 @@ contentful.createClient({
...
-</table>
+You can also combine client chains:
+```js
+const entries = await client.withoutLinkResolution.withAllLocales.getEntries()
+```
 
+The default behaviour didn't change, you can still do
+```js
+const entries = await client.getEntries()
+```
```
[the remark this answers is not in the document]
Comment (marcolink) [reply README.md:305]: @damienxy to what would you refer here?

File: README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (ruderngespra) [reply README.md]: Looks great :)

**Review by ruderngespra — APPROVED** [review]

**Pull request merged**